packages/grid/src/vaadin-grid-styling-mixin.d.ts

```diff
@@ -32,11 +37,39 @@ export declare class StylingMixinClass<TItem> {
    */
   cellClassNameGenerator: GridCellClassNameGenerator<TItem> | null | undefined;
 
+  /**
+   * A function that allows generating CSS `part` names for grid cells in Shadow DOM based
+   * on their row and column, for styling from outside using the `::part()` selector.
+   *
+   * The return value should be the generated part name as a string, or multiple part names
+   * separated by whitespace characters. This
+   *
+   * Receives two arguments:
+   * - `column` The `<vaadin-grid-column>` element (`undefined` for details-cell).
+   * - `model` The object with the properties related with
+   *   the rendered item, contains:
+   *   - `model.index` The index of the item.
+   *   - `model.item` The item.
+   *   - `model.expanded` Sublevel toggle state.
+   *   - `model.level` Level of the tree represented with a horizontal offset of the toggle button.
+   *   - `model.selected` Selected state.
+   *
+   */
+  cellPartNameGenerator: GridCellPartNameGenerator<TItem> | null | undefined;
+
   /**
    * Runs the `cellClassNameGenerator` for the visible cells.
    * If the generator depends on varying conditions, you need to
    * call this function manually in order to update the styles when
    * the conditions change.
    */
   generateCellClassNames(): void;
```

Comment:
Should `generateCellClassNames` and  `cellClassNameGenerator` also be deprecated if we're deprecating `GridCellClassNameGenerator`

Reply:
Done.